ovs/extensions/migration/migration/albamigrator.py

```diff
@@ -49,6 +53,30 @@ def migrate(previous_version, master_ips=None, extra_ips=None):
 
         # From here on, all actual migration should happen to get to the expected state for THIS RELEASE
         if working_version < AlbaMigrator.THIS_VERSION:
-            pass
+            try:
+                from ovs.dal.lists.albabackendlist import AlbaBackendList
+                from ovs.extensions.generic.configuration import Configuration
+                from ovs.lib.alba import AlbaController
+
+                AlbaMigrator._logger.info('Starting migrations...')
+
+                # This could be handled by out of band migrations, but since post-update restarts the services,
+                # putting this in the out of band migration code would result in the services being restarted before
+                # having updated this setting (See wrong order)
+                # Correct order: This migration code --> Post-update code --> Restart services
+                # Wrong order  : Post-update code --> Restart services --> Out of band migration making changes to configuration
+                for alba_backend in AlbaBackendList.get_albabackends():
+                    config_key = '{0}/maintenance/config'.format(AlbaController.CONFIG_ALBA_BACKEND_KEY.format(alba_backend.guid))
+                    if Configuration.exists(key=config_key):
+                        config = Configuration.get(key=config_key)
+                        if 'multicast_discover_osds' not in config:
+                            config['multicast_discover_osds'] = False
+                            Configuration.set(key=config_key, value=config)
+                            AlbaMigrator._logger.debug('Updated multi-cast setting for ALBA Backend {0}'.format(alba_backend.name))
+                AlbaMigrator._logger.info('Finished migrations')
+            except:
+                AlbaMigrator._logger.exception('Error occurred while executing the ALBA migration code')
```

Comment:
It looks like in case of an exception this operation will not be retried, nor fail the update.
You may want to change that behaviour. Similar comment for the asd config migration code.